PublishParasoftTestResults/xsl/sarif.xsl

```diff
@@ -404,12 +405,79 @@
                 </xsl:if>
             </xsl:when>
             <xsl:otherwise>
-                <xsl:value-of select="@locFile" /><xsl:text>"</xsl:text>
+                <xsl:choose>
+                    <xsl:when test="$isCPPProReport">
+                        <!-- For cppTest professional report, "prjModule" attribute is not present.
+                            Project name presents twice in the full source path and needs to be removed.-->
+                        <xsl:value-of select="$locationMap(@locFile)" /><xsl:text>"</xsl:text>
+                    </xsl:when>
+                    <xsl:when test="/ResultsSession/@toolId = 'dottest'">
+                        <!-- For DotTest report, project name prefix is missing in "resProjPath" of "Loc".
+                               As a result, the root folder will be removed from full source path. -->
+                        <xsl:value-of select="concat('/', substring-after(@locFile, '/'))"/><xsl:text>"</xsl:text>
+                    </xsl:when>
+                    <xsl:otherwise>
+                        <!-- For Jtest and cppTest standard reports, use "resProjPath" in "Loc". -->
+                        <xsl:call-template name="processLocation">
+                            <xsl:with-param name="locNode" select="$locNode"/>
+                            <xsl:with-param name="locFile" select="@locFile"/>
+                        </xsl:call-template>
+                    </xsl:otherwise>
+                </xsl:choose>
             </xsl:otherwise>
         </xsl:choose>
 
         <xsl:text> }</xsl:text>
     </xsl:template>
+
+    <xsl:variable name="locationMap" as="map(xs:string, xs:string)" >
+        <xsl:if test="$isCPPProReport">
+            <xsl:map>
+                <xsl:for-each select="/ResultsSession/CodingStandards/TestedFilesDetails/Total//Res[@loc]">
+                    <xsl:call-template name="processCppTestProLocation">
```

Comment:
Please remind me why can't we just use:
`<xsl:map-entry key="concat('', @loc)" select="concat('/', @name)"/>`

That `loc` attributes should be generated with the parents `name` right?
![image](https://user-images.githubusercontent.com/35082985/232929544-7101c4c5-c32b-4c44-bc6e-88147d197ff8.png)